app/routes/_layout+/calendar.tsx

```diff
@@ -0,0 +1,220 @@
+
+export function links() {
+  return [{ rel: "stylesheet", href: calendarStyles }];
+}
+
+export const handle = {
+  breadcrumb: () => <Link to="/calendar">Calendar</Link>,
+};
+
+// Loader Function to Return Bookings Data
+export const loader = async ({ request, context }: LoaderFunctionArgs) => {
+  const authSession = context.getSession();
+  const { userId } = authSession;
+
+  try {
+    await requirePermission({
+      userId,
+      request,
+      entity: PermissionEntity.booking,
+      action: PermissionAction.read,
+    });
+
+    const header = {
+      title: `Calendar`,
+    };
+
+    const currentDate = new Date();
+    const currentMonth = currentDate.toLocaleString("default", {
+      month: "long",
+    });
+    const currentYear = currentDate.getFullYear();
+
+    const title = `${currentMonth} ${currentYear}`;
+
+    return json(data({ header, title }));
+  } catch (cause) {
+    const reason = makeShelfError(cause);
+    throw json(error(reason), { status: reason.status });
+  }
+};
+export const meta: MetaFunction<typeof loader> = ({ data }) => [
+  { title: appendToMetaTitle(data?.header.title) },
+];
+
+// Calendar Component
+const Calendar = () => {
+  const { title } = useLoaderData<typeof loader>();
+  const [_error, setError] = useState<string | null>(null);
+  const [calendarTitle, setCalendarTitle] = useState(title);
+  const calendarRef = useRef<FullCalendar>(null);
+  const ripple = useRef<HTMLDivElement>(null);
+
+  const handleNavigation = (navigateTo: any) => {
+    const calendarApi = calendarRef.current?.getApi();
+    if (navigateTo == "prev") {
+      calendarApi?.prev();
+    } else if (navigateTo == "next") {
+      calendarApi?.next();
+    } else if (navigateTo == "today") {
+      calendarApi?.gotoDate(new Date());
+    }
+    updateTitle();
+  };
+
+  const updateTitle = () => {
+    const calendarApi = calendarRef.current?.getApi();
+    if (calendarApi) {
+      const currentDate = calendarApi.getDate();
+      const currentMonth = currentDate.toLocaleString("default", {
+        month: "long",
+      });
+      const currentYear = currentDate.getFullYear();
+      setCalendarTitle(`${currentMonth} ${currentYear}`);
+    }
+  };
+
+  const toggleSpinner = useCallback(
+    (state: any) => {
+      if (ripple.current) {
+        if (state) {
+          ripple.current.classList.remove("hidden");
+        } else {
+          ripple.current.classList.add("hidden");
+        }
+      }
+    },
+    [ripple]
+  );
+
+  type BookingStatus =
+    | "DRAFT"
+    | "ARCHIVED"
+    | "CANCELLED"
+    | "RESERVED"
+    | "ONGOING"
+    | "OVERDUE"
+    | "COMPLETE";
+
+  const statusClassesOnHover: Record<BookingStatus, string> = {
+    DRAFT: "!bg-gray-100",
+    ARCHIVED: "!bg-gray-100",
+    CANCELLED: "!bg-gray-100",
+    RESERVED: "!bg-blue-100",
+    ONGOING: "!bg-purple-100",
+    OVERDUE: "!bg-warning-100",
+    COMPLETE: "!bg-success-100",
+  };
+
+  const handleEventMouseEnter = (info: any) => {
```

Comment:
Do not use type `any`. Pretty sure you can get the type of info from FullCalendar types.